internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -128,6 +138,10 @@ func (r *BMACReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return result.Result()
 	}
 
+	if result.Complete() {
```

Comment:
I think the word complete could be confusing. It could be interpreted as the complete set of steps in the main Reconcile function is done. But, the only place where reconcileComplete.complete is set to true is in reconcileBMH. And it is really used to stop the reconcile from proceeding to steps that are currently unnecessary. I think you mention this in the git commit message, but I think it would be useful to mention this in a comment in the code.

Reply:
I spent a significant amount of minutes trying to find a good name for this and it was just after reading your code that the following names came top mind (which I think would be more descriptive):

- `Stop`
- `Interrupt`

I will probably go with the former 'cause it is shorter.